internal/controllers/mic_reconciler.go

```diff
@@ -0,0 +1,336 @@
+		case v1.PodSucceeded:
+			logger.Info("successful pod, updating image status to ImageExists")
+			mrhi.micHelper.SetImageStatus(micObj, image, kmmv1beta1.ImageExists)
+			podsToDelete = append(podsToDelete, p)
+		}
+	}
+	// patch the status in the MIC object
+	err := mrhi.client.Status().Patch(ctx, micObj, patchFrom)
+	if err != nil {
+		return fmt.Errorf("failed to patch the status of mic %s: %v", micObj.Name, err)
+	}
+
+	errs := make([]error, 0, len(podsToDelete))
+	for _, pod := range podsToDelete {
+		err = mrhi.podHelper.deletePod(ctx, &pod)
+		errs = append(errs, err)
+	}
+
+	return errors.Join(errs...)
+}
+
+func (mrhi *micReconcilerHelperImpl) updateStatusByMBSC(ctx context.Context, micObj *kmmv1beta1.ModuleImagesConfig) error {
+	mbsc, err := mrhi.mbscHelper.GetMBSC(ctx, micObj.Name, micObj.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get ModuleBuildSignConfig object %s/%s: %v", micObj.Namespace, micObj.Name, err)
+	}
+
+	if mbsc == nil {
+		return nil
+	}
+
+	patchFrom := client.MergeFrom(micObj.DeepCopy())
+	for _, imageState := range mbsc.Status.ImagesStates {
+		imageSpec := mrhi.micHelper.GetModuleImageSpec(micObj, imageState.Image)
+		if imageSpec == nil {
+			// image not found in spec, ignore
+			continue
+		}
+		micImageStatus := kmmv1beta1.ImageExists
+		if imageState.Status == kmmv1beta1.ImageBuildFailed {
+			micImageStatus = kmmv1beta1.ImageDoesNotExist
+		}
+		mrhi.micHelper.SetImageStatus(micObj, imageState.Image, micImageStatus)
+	}
+
+	return mrhi.client.Status().Patch(ctx, micObj, patchFrom)
+}
+
+func (mrhi *micReconcilerHelperImpl) processImagesSpecs(ctx context.Context, micObj *kmmv1beta1.ModuleImagesConfig, pullPods []v1.Pod) error {
+	errs := make([]error, len(micObj.Spec.Images))
+	for _, imageSpec := range micObj.Spec.Images {
+		imageState := mrhi.micHelper.GetImageState(micObj, imageSpec.Image)
+
+		// image State does not exists either new image or pull pod is still running
+		switch imageState {
+		case "":
+			// image State is not set: either new image or pull pod is still running
+			if mrhi.podHelper.getPullPodForImage(pullPods, imageSpec.Image) == nil {
+				// no pull pod- create it, otherwise we wait for it to finish
+				err := mrhi.podHelper.createPullPod(ctx, &imageSpec, micObj)
+				errs = append(errs, err)
+			}
+		case kmmv1beta1.ImageDoesNotExist:
+			if imageSpec.Build == nil && imageSpec.Sign == nil {
+				break
+			}
+			fallthrough
+		case kmmv1beta1.ImageNeedsBuilding:
+			// image needs to be built - patching MBSC
+			mbscObj := &kmmv1beta1.ModuleBuildSignConfig{
+				ObjectMeta: metav1.ObjectMeta{Name: micObj.Name, Namespace: micObj.Namespace},
+			}
+			_, err := controllerutil.CreateOrPatch(ctx, mrhi.client, mbscObj, func() error {
+				mrhi.mbscHelper.SetModuleImageSpec(mbscObj, &imageSpec)
+				mbscObj.Spec.ImageRepoSecret = micObj.Spec.ImageRepoSecret
+				return controllerutil.SetOwnerReference(micObj, mbscObj, mrhi.scheme)
+			})
```

Comment:
How about we move all of this to the MBSC interface since we already have this package and this logic doesn't contain any logic outside of the MBSC scope?